### ISSUE TITLE
mupdf: Direction set as default does not apply

### ISSUE THREAD

**Issue by mergen3107** [koreader/koreader#10201]
* KOReader version: 2023.01-100
* Device: Onyx Boox Kon-Tiki 2 (Android 10)

#### Issue
Direction setting is not applying its user-set default option. 

#### Steps to reproduce
Open a pdf file. 
Go to Zoom mode: Columns, 1.9 Columns. 
Tap to set Direction to second option (Top to Bottom, Left to Right). 
Then long tap to set it as default as well. 
Go to Top menu - 2nd Tab - Doc settings - Reset document settings. 
Here my default zoom is content, crop level is auto, so I go and set the Zoom mode to column again. 
Try to advance - notice that the Direction is as the first option, although it shows on the bottom menu that it is “set” to second option (as well as the long tap reveals that this is the “default option). 

##### `crash.log` (if applicable)
I hope this is reproducible on your end. 

I could do logs, but later (on mobile right now)

**Comment by mergen3107** [comment]
For now I have this workaround:
I set “Spread” gesture to set Zoom mode: Columns and Direction: to the second option. 

However, the bottom menu Columns/Rows/Percentage selection still yields into confusion.

**Comment by NiLuJe** [comment]
There's another semi-recent open-then-closed issue about this (possibly. kopt and readerzoom are hell, so, who knows) or something similar at least.

Closed because we (well, @hius07; I myself need to be heavily inebriated to be able to deal with this mess ;p) couldn't reproduce it. It may or may not be related to old global or document defaults; because there's an insane amount of redirection and legacy code in there that's very very easy to break.

TL;DR: Check on a stock install with no config (e.g., an emu build).

**Comment by NiLuJe** [comment]
Found it: #9321

**Comment by mergen3107** [comment]
Is that `kopt_zoom_direction` in settings.reader.lua?
It currently says “6”

**Comment by NiLuJe** [comment]
Probably, but that's unlikely to be the culprit, as that one doesn't suffer from as much legacy code burden as the rest.
My shot in the dark would be something along the lines of *aznother* readerzooming stuff (possibly a legacy one) being badly interpreted in a way that actually ends up bypassing or ingoring the zoom direction stuff (because that only makes sense in *some* zoom mode combinations; and there's a billion of those).

**Comment by mergen3107** [comment]
I can confirm this issue is reproducible on the fresh Linux emulator.
Steps are exactly the same.

**Comment by mergen3107** [comment]
I couldn't find the crash.log in the dev build, so here is simultaneous terminal output (dev logs ON):

[zoom_direction1.txt](https://github.com/koreader/koreader/files/10953100/zoom_direction1.txt)

1) Open KOReader via `./kodev run`
2) It opens the User guide. Currently with zoom set to `content`.
3) Reset document settings.
4) Set columns. Select second direction.
5) Long tap and select that as default.
6) Reset document settings again.
7) Set columns, then scroll - wrong direction. Although settings "says" it is the second one.
8) Tap the second direction again - now it works ok.

**Comment by mergen3107** [comment]
Thank you @hius07 ! 
Will test tomorrow

**Comment by mergen3107** [comment]
Thanks again @hius07! Works absolute charms for me. I removed the extra gesture to select the direction, don’t need it anymore :D 

Later going to set RTL directions too. Like does it start at the right corner and so on (didn’t have time just for this ><)
